mlflow/store/artifact/cloud_artifact_repo.py

```diff
@@ -212,20 +213,30 @@ def _parallelized_download_from_cloud(self, file_size, remote_file_path, local_p
             )
 
             if failed_downloads:
-                new_cloud_creds = self._get_read_credential_infos([remote_file_path])[0]
-                new_signed_uri = new_cloud_creds.signed_uri
-                new_headers = self._extract_headers_from_credentials(new_cloud_creds.headers)
+                num_retries = 3
                 for chunk in failed_downloads:
-                    _logger.warning(
-                        f"Retrying download of chunk {chunk.index} of {remote_file_path}"
-                    )
-                    download_chunk(
-                        range_start=chunk.start,
-                        range_end=chunk.end,
-                        headers=new_headers,
-                        download_path=local_path,
-                        http_uri=new_signed_uri,
-                    )
+                    new_cloud_creds = self._get_read_credential_infos([remote_file_path])[0]
+                    new_signed_uri = new_cloud_creds.signed_uri
+                    new_headers = self._extract_headers_from_credentials(new_cloud_creds.headers)
+                    _logger.warning(f"Retrying download of chunk {chunk} for {remote_file_path}")
+                    for retry in range(num_retries):
+                        try:
+                            _logger.warning(f"Attempt {retry + 1}/{num_retries}")
+                            download_chunk(
+                                range_start=chunk.start,
+                                range_end=chunk.end,
+                                headers=new_headers,
+                                download_path=local_path,
+                                http_uri=new_signed_uri,
+                            )
+                            _logger.warning(
+                                f"Successfully downloaded chunk {chunk} for {remote_file_path}"
+                            )
+                            break
+                        except Exception:
+                            if retry == num_retries - 1:
+                                raise
+                        time.sleep(1)
```

Comment:
Similar to my comment above about retry count, what are your thoughts on a configurable sleep duration for retries? Since we're doing a loop over each partition transfer and only waiting 1s between each try, there may be considerations in the R2 provider where a longer wait may increase the probability of success on aggregate based on observations once this is released. 

Do you think a configurable value for this might be worthwhile?